### DISCUSSION TITLE
CFS Apps under Linux: Process or Thread?

### DISCUSSION THREAD

**Discussion by dtran11** [nasa/cFS#722] [Q&A]
I have about 10 custom apps under Linux. Does the CFE spawn my apps into individual processes (with separate memory region) or are they threads of a single process? Thanks

**Comment by irowebbn** [comment]
If you are using the provided OSAL layer, apps are spawned using POSIX threads. See https://github.com/nasa/osal/blob/1a8823f2dc93da2bcb3696c92c7aee4f512ed4f7/src/os/posix/src/os-impl-tasks.c#L450

**Reply by dtran11** [reply]
Thanks for the info. Has there been effort to use processes instead of threads on Linux so that one bad thread cannot bring all other apps down?

**Reply by irowebbn** [reply]
I don't know. Judging by a quick search through the public issues and pull requests, it doesn't seem like it.

**Reply by skliper** [reply]
Mixed trust/confidence and or memory/process partitioning is more often handled by running separate executables (and SBN or similar if needed).
